src/topology/mod.rs

```diff
@@ -71,29 +70,16 @@ pub async fn start_validated(
     Some((running_topology, abort_rx))
 }
 
+// TODO: inline this
```

Comment:
Got an issue for this? Or is it for this PR still?

Reply:
I started down this path and it lead to a bit more than I want to add to this PR, so I'll just rename it to something more appropriate. We can continue the refactoring later.